app/(main)/create-market/page.tsx

```diff
@@ -0,0 +1,671 @@
+import { useAccount, useConfig, useWriteContract } from 'wagmi';
+import {
+  formatDateTime,
+  formatEtherWithFixedDecimals,
+  formatValueWithFixedDecimals,
+} from '@/utils';
+import { erc20Abi, formatEther, parseEther } from 'viem';
+import { getUnixTime, isPast } from 'date-fns';
+import { useUnsupportedNetwork } from '@/hooks';
+import { readContract, simulateContract, waitForTransactionReceipt } from 'wagmi/actions';
+import {
+  ORACLE_ADDRESS,
+  REALITY_ETH_CONTRACT_ADDRESS,
+  REALITY_ETH_TIMEOUT,
+  SINGLE_SELECT_TEMPLATE_ID,
+} from '../../../hooks/contracts/realityeth';
+import { KLEROS_ARBITRATOR_ADDRESS } from '../../../hooks/contracts/kelros';
+import RealityABI from '@/abi/reality.json';
+import ConditionalTokensABI from '@/abi/conditionalTokens.json';
+import MarketMakerFactoryABI from '@/abi/marketMakerFactory.json';
+import {
+  CONDITIONAL_TOKEN_CONTRACT_ADDRESS,
+  MARKET_MAKER_FACTORY_ADDRESS,
+} from '../../../hooks/contracts';
+import { calcDistributionHint } from '../../../utils/liquidity';
+import { ModalId, useModal } from '../../../context';
+
+type InputOutcome = {
+  name: string;
+  percentage: number;
+};
+
+const defaultInputOutcomes: InputOutcome[] = [
+  { name: '', percentage: 50 },
+  { name: '', percentage: 50 },
+];
+const tokenList = [WXDAI, SDAI];
+
+const LP_FEE = parseEther('0.02');
+const getDefaultDateTime = () => {
+  const now = new Date();
+
+  // Format as 'YYYY-MM-DDTHH:mm'
+  return now.toISOString().slice(0, 16);
+};
+
+export default function CreateMarketPage() {
+  const { address, isDisconnected } = useAccount();
+  const { openModal } = useModal();
+
+  const [question, setQuestion] = useState('');
+  const [category, setCategory] = useState('');
+  const [amount, setAmount] = useState('');
+  const [outcomes, setOutcomes] = useState<InputOutcome[]>(defaultInputOutcomes);
+  const [closingDate, setClosingDate] = useState(getDefaultDateTime());
+  const [marketToken, setMarketToken] = useState<Token>(SDAI);
+  const unsupportedNetwork = useUnsupportedNetwork();
+
+  const [percentagesError, setPercentagesError] = useState<Record<number, string> | null>(
+    null
+  );
+  const [dateTimeError, setDateTimeError] = useState<string | null>(null);
+  const [error, setError] = useState<string | null>(null);
+
+  const { data: balance } = useReadBalanceOf({
+    address,
+    tokenAddress: marketToken.address,
+  });
+
+  useEffect(() => {
+    if (!question) setError('Enter market quesiton');
+    else if (outcomes.findIndex(outcome => !outcome.name) !== -1)
+      setError('Enter outcome names');
+    else if (!category) setError('Enter category');
+    else if (!amount || +amount === 0) setError('Enter amount');
+    else if (dateTimeError) setError('Enter valid closing date');
+    else if (percentagesError) setError('Enter valid outcomes');
+    else setError(null);
+  }, [amount, category, dateTimeError, outcomes, percentagesError, question]);
+
+  const maxBalance = useCallback(() => {
```

Comment:
Is there a significant benefit in wrapping all these state setters with the `useCallback` hook? I just tried to profile a bit this component page with and without these hooks, and by just removing the usage of `useCallback` we gain +8% perfo in Lighthouse

With the hooks in
<img width="734" alt="Screenshot 2024-12-24 at 10 00 48 AM" src="https://github.com/user-attachments/assets/5d577bcc-1549-4548-8d64-9c104d5e51be" />

Without the hooks
<img width="737" alt="Screenshot 2024-12-24 at 11 06 01 AM" src="https://github.com/user-attachments/assets/625726f8-ea45-48fd-a3f4-120e508bb89b" />